server/protocols/webadmin/webadmin-core/src/main/java/org/apache/james/webadmin/routes/HealthCheckRoutes.java

```diff
@@ -152,6 +152,14 @@ private int getCorrespondingStatusCode(ResultStatus resultStatus) {
     private void logFailedCheck(Result result) {
         switch (result.getStatus()) {
         case UNHEALTHY:
+            if (result.getError().isPresent()) {
+                LOGGER.error("HealthCheck failed for {} : {}",
+                    result.getComponentName().getName(),
+                    result.getCause().orElse(""),
+                    result.getError().get());
+                break;
```

Comment:
Can we have a il {...} else { ... } instead of this hard to understand flow control?

Reply:
Also handle the error logging also when DEGRADED